gateway/engine/core/src/main/java/io/apiman/gateway/engine/impl/EngineImpl.java

```diff
@@ -64,6 +67,7 @@ public EngineImpl(final IRegistry registry, final IPluginRegistry pluginRegistry
         
         policyFactory.setPluginRegistry(pluginRegistry);
         metrics.setComponentRegistry(componentRegistry);
+        initializeMetric(metrics);
```

Comment:
Just on the factories and registries.  Specifically, these:

https://github.com/apiman/apiman/blob/master/gateway/engine/core/src/main/java/io/apiman/gateway/engine/impl/EngineImpl.java#L39-L44